pkg/activator/throttler_test.go

```diff
@@ -128,6 +130,64 @@ func TestThrottler_UpdateCapacity(t *testing.T) {
 	}
 }
 
+func TestThrottler_ManyActivatorEndpoints(t *testing.T) {
+	const updatePollInterval = 10 * time.Millisecond
+	const updatePollTimeout = 3 * time.Second
+	activatorCount := 1
+	revConcurrency := 10
+
+	ep := &corev1.Endpoints{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      testRevision,
+			Namespace: testNamespace,
+		},
+		Subsets: endpointsSubset(1, 1),
+	}
+	activatorEp := &corev1.Endpoints{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      K8sServiceName,
+			Namespace: system.Namespace(),
+		},
+		Subsets: endpointsSubset(1, activatorCount),
+	}
+
+	fake := kubefake.NewSimpleClientset(ep)
+	informer := kubeinformers.NewSharedInformerFactory(fake, 0)
+	endpoints := informer.Core().V1().Endpoints()
+	endpoints.Informer().GetIndexer().Add(ep)
+
+	stopCh := make(chan struct{})
+	defer close(stopCh)
+	controller.StartInformers(stopCh, endpoints.Informer())
+	fake.CoreV1().Endpoints(activatorEp.Namespace).Create(activatorEp)
+
+	throttler := getThrottler(
+		defaultMaxConcurrency,
+		revisionLister(testNamespace, testRevision, v1beta1.RevisionContainerConcurrencyType(revConcurrency)),
+		endpoints,
+		nil,
+		TestLogger(t),
+		initCapacity,
+	)
+
+	wantCapacity := revConcurrency / activatorCount
+
+	err := wait.PollImmediate(updatePollInterval, updatePollTimeout, func() (bool, error) {
+		err := throttler.UpdateCapacity(revID, 1)
+		if err != nil {
+			return false, err
+		}
+
+		if got := throttler.breakers[revID].Capacity(); got != wantCapacity {
+			return false, nil
+		}
+		return true, nil
+	})
+	if err != nil {
```

Comment:
Could you follow the pattern from the ReactToEndpoint test and check the condition again after the loop to be able to log the got/want values for better insights?

Reply:
Added the check for the values